### PR TITLE
Restyle vault creation form

### LINKED ISSUE #451: Build Vaults List Page: Create Vault
### Specification
Finish the "Create New Vault" modal that can mint/create a new vault. Allow the user to select an asset or specify a custom asset to use. 

https://www.figma.com/file/wXF7xGVv1j2SqwWfgbamS8/Coordinape-Design-v8?node-id=2136%3A24649

Still to do:
- [x] Handle custom asset: When a custom asset is chosen, the form needs to read the asset's contract to determine it's symbol and number of decimal places. 
- [x] add final description
- [x] finalize the list of assets that should be shown

### PR DESCRIPTION
**Pull request by levity**
Closes #451:
- implements new design for vault creation form
- reads & validates symbol and decimal places from custom asset's contract before form submission
- removes the old form modal and supporting code
- adds some tests

Also:
- adds a modal component based on Radix UI's Dialog
- removes old vault contract wrapper code
- moves the list of supported tokens to services/contracts, & checks it against deploymentInfo.json
- displays the token symbol correctly for custom assets on DistributePage

### CODE REVIEW

File: src/pages/VaultsPage/CreateForm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  fireEvent,
+  render,
+  screen,
+  waitFor,
+} from '@testing-library/react';
+import 'hooks/gql';
+import { TestWrapper } from 'utils/testing';
+import { CreateForm } from './CreateForm';
+
+jest.mock('hooks/gql', () => {
```
Comment (levity) [inline src/pages/VaultsPage/CreateForm.test.tsx:12]: very glad this mock was so easy to set up

Comment (levity) [inline src/pages/VaultsPage/CreateForm.test.tsx:12]: very glad this mock was easy to set up

File: src/services/contracts.ts [hunk 2]
```diff
@@ -14,24 +14,31 @@ import {
...
+  'ApeDistributor',
+];
+
+export const supportedChainIds: number[] = Object.entries(deploymentInfo)
+  .filter(([, contracts]) => requiredContracts.every(c => c in contracts))
+  .map(x => Number(x[0]));
+
+const symbols = ['DAI', 'USDC', 'USDT', 'YFI'] as const;
+export const AssetEnum = z.enum(symbols);
+export type Asset = z.infer<typeof AssetEnum>;
```
Comment (teeolendo) [inline src/services/contracts.ts:35]: Would it be easier to just instantiate a string enum and use it as a type?
```
export const Asset = {
  DAI = "DAI",
  USDC = "USDC"
  ...
}
```
Don't see why we need use to zod. Happy to be schooled ofc.

File: .eslintrc [hunk 3]
```diff
@@ -113,7 +113,6 @@
           }
         ],
         "pathGroupsExcludedImportTypes": [],
-        "newlines-between": "always",
```
[the remark this answers is not in the document]
Comment (levity) [reply .eslintrc:116]: how would you feel about switching to a simpler model, say, just alphabetical order without groups?

[the remark this answers is not in the document]
Comment (levity) [reply .eslintrc:116]: in any case, i'll revert this for now & we can have a proper discussion with the team

[the remark this answers is not in the document]
Comment (levity) [reply .eslintrc:116]: just found another way to avoid being irritated by this rule, which is to add this to my vscode settings.json:

```
```

**Pull request merged**